Add info about soft and hard deletes (#2249)
astronomer/docs: software/configure-deployment.md

## Delete a Deployment

You can delete an Airflow Deployment using the **Delete Deployment** button at the bottom of the Deployment's **Settings** tab. 

When you delete a Deployment, you delete your Airflow webserver, scheduler, metadata database, and deploy history, and you lose any configurations set in the Airflow UI. 

> By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database.

Reviewer: Additionally the cleanup job can be tweaked via the astronomer helm chart

`astronomer.houston.cleanupDeployments`

```
# Cleanup deployments that have been soft-deleted in Houston
  # This runs as a CronJob
  cleanupDeployments:
    # Enable cleanup CronJob
    enabled: true

    # Default here is to run at midnight every night https://crontab.guru/#0_0_*_*_*
    schedule: "0 0 * * *"

    # Cleanup deployments older than this many days
    olderThan: 14
```

> By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database.

Reviewer: lgtm

## Delete a Deployment

You can delete an Airflow Deployment using the **Delete Deployment** button at the bottom of the Deployment's **Settings** tab. 

When you delete a Deployment, you delete your Airflow webserver, scheduler, metadata database, and deploy history, and you lose any configurations set in the Airflow UI. 

> By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database. 14 days after your Deployment's soft delete, Astronomer automatically runs a hard delete cronjob that deletes any persisting values from your soft delete.

Reviewer 2: ```suggestion
By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database. 14 days after your Deployment's soft delete, Astronomer automatically runs a hard delete cronjob that deletes any values that remained after your soft delete.
```

Author: added the most recent suggestion only

> By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database. 14 days after your Deployment's soft delete, Astronomer automatically runs a hard delete cronjob that deletes any persisting values from your soft delete.

Reviewer 2: ```suggestion
By default, Astro performs a _soft delete_ when you delete a Deployment. After you delete a Deployment, your Astronomer database, the corresponding `Deployment` record receives a `deletedAt` value and continues to persist until permanently deleted through a _hard delete_. A hard delete includes both the Deployment's metadata database and the Deployment entry in your Astronomer database. 14 days after your Deployment's soft delete, Astronomer automatically runs a hard delete cronjob that deletes any values that remained after your soft delete.
```

:::tip

Astronomer recommends regularly doing a database audit to confirm that you hard delete databases.

:::

### Automate Deployment deletion clean up 

Astronomer runs a cronjob to hard delete the deleted Deployment's metadata database and Deployment entry in your Astronomer database at midnight on a specified day. You can enable whether this cronjob runs or not, how many days after your soft delete to run the cronjob, and what time of day to run the cronjob by editing `astronomer.houston.cleanupDeployments` in your Astronomer Helm chart.

> The following example shows the default settings you can find in your Helm chart.

Reviewer 2: ```suggestion
The following is an example of how you might configure the cronjob in your Helm chart: 
```

Author: added the most recent suggestion only

> The following example shows the default settings you can find in your Helm chart.

Reviewer 2: ```suggestion
The following is an example of how you might configure the cronjob in your Helm chart: 
```

Author: ah thank you for this. I was struggling with the phrasing here.

> ```

Reviewer 2: ```suggestion
```yaml
```

# Cleanup deployments that have been soft-deleted
  # This clean up runs as a CronJob
  cleanupDeployments:
    # Enable the clean up CronJob
    enabled: true

    # Default time for the CronJob to run https://crontab.guru/#0_0_*_*_*
    schedule: "0 0 * * *"

    # Number of days after the Deployment deletion to run the CronJob
    olderThan: 14
```

### Hard delete a Deployment

To reuse a custom release name given to an existing Deployment after a soft delete but before Astronomer automatically cleans up any persisting Deployment records, you need to hard delete both the Deployment's metadata database and the Deployment's entry in your Astronomer database. 

1. Enable hard delete as an option at the platform level. To enable this feature, set `astronomer.houston.config.deployments.hardDeleteDeployment: true` in your `config.yaml` file and push the changes to your platform as described in [Apply a config change](apply-platform-config.md).

> 2. You can delete a Deployment with the UI or CLI.

Reviewer 2: ```suggestion
2. Hard delete a Deployment with the Software UI or Astro CLI.
```

>   - **In the UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 

Reviewer 2: ```suggestion
  - **In the Software UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 
```

Author: added the most recent suggestion only

>   - **In the UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 

Reviewer 2: ```suggestion
  - **Software UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 
```

Author: added the most recent suggestion only

>   - **In the UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 

Reviewer 2: ```suggestion
  - **Software UI:** Go to the Deployment's **Settings** tab and select **Delete Deployment**. Then, select the **Hard Delete?** checkbox before confirming **Delete Deployment**. 
  
    ![Hard delete checkbox](/img/software/hard-delete.png)
    
```

>   - **Astro CLI:** Alternatively, you can run `astro deployment delete --hard`.

Reviewer 2: ```suggestion
  - **Astro CLI:** Run `astro deployment delete --hard`.
```

This action permanently deletes all data associated with a Deployment, including the database and underlying Kubernetes resources.

> ![Hard delete checkbox](/img/software/hard-delete.png)

Reviewer 2: ```suggestion
```
